pkg/shard/mock_shard.go

```diff
@@ -0,0 +1,40 @@
+package shard
+
+import (
+	"crypto/tls"
+	"github.com/jackc/pgx/v5/pgproto3"
+	"github.com/pg-sharding/spqr/pkg/config"
+	"github.com/pg-sharding/spqr/pkg/conn"
+	"github.com/pg-sharding/spqr/pkg/models/kr"
+	"github.com/pg-sharding/spqr/pkg/txstatus"
+)
+
+type mockShard struct {
```

Comment:
please use `MockShard` from `mock` package

Reply:
I can't because it causes import cycle

Reply:
anyway you should use mockgen.
What is the import cycle are you struggling with?

Reply:
> What is the import cycle are you struggling with?

package github.com/pg-sharding/spqr/pkg/shard
	imports github.com/pg-sharding/spqr/pkg/mock/shard
	imports github.com/pg-sharding/spqr/pkg/shard: import cycle not allowed in test

Reply:
Consider changing package name in your test to shard_test